### DISCUSSION TITLE
Retrieving the latest version of applications via a Package Bundle

### DISCUSSION THREAD

**Discussion by ninbura** [Devolutions/UniGetUI#2697] [Q&A] [answered]
Is there a way to make packages within a bundle retrieve, display, and install the latest version of said package at install?

I'm noticing that when I create a package bundle, the application version of apps in said bundle are determined at the time when I added the package to the bundle.

For example, if I add VS Code to a bundle, wait 5 months, open the bundle up on another computer, the version of VS Code in the bundle is now 5 months old. It doesn't attempt to get the latest version of that app, it prioritizes installing the exact version of the app that was specified when the app was added to the bundle.

I've found that if I open the `.json` file, and manually remove the `Version` value, then it will retrieve the latest version when bundle is loaded at a later date. 
```json
    {
      "Id": "Rufus.Rufus",
      "Name": "Rufus",
      "Version": "", <---------------------------- manually removed version number
      "Source": "winget",
      "ManagerName": "Winget",
      "InstallationOptions": {
        "SkipHashCheck": false,
        "InteractiveInstallation": false,
        "RunAsAdministrator": false,
        "Architecture": "",
        "InstallationScope": "",
        "CustomParameters": [],
        "PreRelease": false,
        "CustomInstallLocation": "",
        "Version": ""
      },
      "Updates": {
        "UpdatesIgnored": false,
        "IgnoredVersion": ""
      }
    },
```

However, this also creates a display issue in the Application Bundle, where no version is listed next to the app in the bundle.
![image](https://github.com/user-attachments/assets/f8d816dd-9061-4204-8df1-5171c12343d0)

This makes this workaround less than ideal. Not to mention it takes a fair amount of time to manually remove the version number from 50+ packages in the `.json` file.

**Comment by marticliment** [comment, marked as answer]
This is something I want to fix. The version will not show (will show "latest" or something) unless a specific version was specified on the Installation Settings

**Reply by ninbura** [reply]
Sweet, at least manually removing the version number in the `.json` compels UniGetUI to install the latest version for each package in the mean time.

I would think that most users would prefer that bundles install the latest available versions of packages by default. You can override the "Version to install" within each package's Installation Options if you want to maintain a specific version within the bundle.

**Reply by ninbura** [reply]
I notice that it now displays "Latest" in the package bundle via the GUI. But when I look at the `.ubundle` file, I still see a specific version in the "Version" property.

Is this just a display change? Or is the version in the `.ubundle` ignored and the latest version is installed?

**Reply by marticliment** [reply]
Yes, since the latest version will be installed. The version will only show when a package is set to install a specific version.
